confidential-data-hub/hub/src/hub.rs

```diff
@@ -5,11 +5,59 @@
 
 use async_trait::async_trait;
 use base64::{engine::general_purpose::STANDARD, Engine};
+use kms::{Annotations, ProviderSettings};
 use secret::secret::Secret;
+use tokio::fs;
 
 use crate::{DataHub, Error, Result};
 
-pub struct Hub {}
+pub struct Hub {
+    /// the get resource provider type. Semantically same as kbc.
+    get_resource_provider: String,
+}
+
+impl Hub {
+    pub async fn new() -> Result<Self> {
+        let get_resource_provider = Self::get_resource_provider().await?;
+        #[cfg(feature = "sev")]
+        {
+            Self::init_sev().await?;
+        }
+        Ok(Self {
+            get_resource_provider,
+        })
+    }
+
+    async fn get_resource_provider() -> Result<String> {
+        let cmdline = fs::read_to_string("/proc/cmdline")
+            .await
+            .map_err(|e| Error::InitializationFailed(format!("read kernel cmdline failed: {e}")))?;
+        let resource_provider = cmdline
+            .split_ascii_whitespace()
+            .find(|para| para.starts_with("agent.aa_kbc_params="))
+            .ok_or(Error::InitializationFailed(
+                "no `agent.aa_kbc_params` provided in kernel commandline!".into(),
+            ))?
+            .split("::")
+            .next()
+            .ok_or(Error::InitializationFailed(
+                "illegal input `agent.aa_kbc_params` format".into(),
+            ))?
+            .to_string();
+        Ok(resource_provider)
+    }
+
+    #[cfg(feature = "sev")]
+    async fn init_sev() -> Result<()> {
+        sev::mount_security_fs().map_err(|e| {
```

Comment:
@fitzthum 
I also add `sev` (s.t. `online_sev_kbc`) codes in this PR. We initially design the CDH with an `auth` layer. This function is an example of `auth` layer, s.t. `Sev-ES Security Module` in the [diagram](https://github.com/confidential-containers/guest-components/blob/main/confidential-data-hub/docs/Architecture.svg). Later PR we will extend this in to a function pointer chain, which must be the first codes to be run in CDH's launch. (Usually gets the credentials/auths used by KMS plugins, that is why we name it `auth`)